src/Sylius/Component/Core/Inventory/Operator/OrderInventoryOperator.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Inventory\Operator;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class OrderInventoryOperator implements OrderInventoryOperatorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function cancel(OrderInterface $order)
+    {
+        if (OrderPaymentStates::STATE_PAID === $order->getPaymentState()) {
+            $this->giveBack($order);
+
+            return;
+        }
+        $this->release($order);
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function hold(OrderInterface $order)
+    {
+        /** @var OrderItemInterface $orderItem */
```

Comment:
```php
foreach ($order->getItems() as $orderItem) {
    $variant = $orderItem->getVariant();

    if ($variant->isTracked()) {
        // ... 
    }
}
```

is used quite often in this class, we might extract it to method returning an array of all tracked order item or a method executing a callback on every tracked order item.

Something like this:

```php
foreach ($this->getTrackedOrderItems($order) as $orderItem) {
    // ...
}
```

Or just use the early `continue` to lower the nesting level:

```php
foreach ($order->getItems() as $orderItem) {
    $variant = $orderItem->getVariant();

    if (!$variant->isTracked()) {
        continue;
    }

    // ...
}
```